openassessment/xblock/static/js/spec/lms/oa_file_upload.js

```diff
@@ -46,7 +49,31 @@ describe("OpenAssessment.FileUploader", function() {
                 fileSize: TEST_FILE.size,
                 fileType: TEST_FILE.type
             }
-        );
+        )
+    });
+
+    it("encodes file names correctly when uploading", function() {
+        // Stub the AJAX call, simulating success
+        var successPromise = $.Deferred(
+            function(defer) { defer.resolve(); }
+        ).promise();
+        spyOn($, 'ajax').and.returnValue(successPromise);
+
+        // Upload a file, with special and non-ASCII characters in the name
+        var specialCharacterFile = { ...TEST_FILE };
+        specialCharacterFile.name = FILE_NAME_SPECIAL_CHARS;
+        fileUploader.upload(TEST_URL, specialCharacterFile);
+
+        // Verify that a PUT request was sent with the right parameters
+        expect($.ajax).toHaveBeenCalledWith({
+            url: TEST_URL,
+            type: 'PUT',
+            data: specialCharacterFile,
+            async: false,
+            processData: false,
+            contentType: 'image/jpeg',
+            headers: { 'Content-Disposition': `attachment; filename*=UTF-8''${(FILE_NAME_SPECIAL_ENCODED)}` },
```

Comment:
we don't really care that these specific characters have been sent , we really just care that the name has been encoded.

on the other hand, if we made the line `headers: { 'Content-Disposition': `attachment; filename*=UTF-8''${fileUploader.encodeRFC5987ValueChars(FILE_NAME_SPECIAL_CHARS)}` },`, we're pretty much just duplicating the functionality in a test, which we also don't want to do. 

ultimately, don't really matter. go with your heart .....unless someone else feels strongly about this

Reply:
Yeah, I'm torn about how to test this, but at least this will break if we make any changes to the encoding and force us to go revisit the test.

Reply:
I like tests that break when something is wrong.